src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse.cs

```diff
@@ -467,7 +467,22 @@ public static class Sse
         /// <summary>
         /// __m128 _mm_set_ps (float e3, float e2, float e1, float e0)
         /// </summary>
-        public static Vector128<float> SetVector128(float e3, float e2, float e1, float e0) => SetVector128(e3, e2, e1, e0);
+        public static unsafe Vector128<float> SetVector128(float e3, float e2, float e1, float e0)
+        {
+            // TODO-CQ Optimize algorithm choice based on benchmarks
+
+            // Zero vector and load e2 et index 0
+            Vector128<float> e2Vector = LoadScalarVector128(&e2);
```

Comment:
@CarolEidt, looking at more existing algorithms, it does look like we will want something that can allow users to use `float`/`double` in these methods (without cost).

It would seem like having an `UnsafeCast` method (or better name) would work well (assuming it also explicitly documents the value of the upper bits as "indeterminate"). -- Maybe we could update `Unsafe.As` to do the right thing, but that would require users to pull in another package to do this stuff...

I had thought about providing additional overloads (such as `UnpackLow(float, float)`), but that still results in "indeterminate" results in some scenarios (such as the two upper bits of the result).

Reply:
> Maybe we could update Unsafe.As to do the right thing

Actually this what are you commenting on are some of the most serious pain points in current API surface. We need easy and low cost way to load from memory and move from register any of the supported numerals into lowest position of the vector:

```C#

int value = Foo();
.... 
Vector<int> vector = int.AsVector();
// this may allow to write
Vector<int> vector = UnapckLow(value.AsVector(), value.AsVector());
```

Reply:
We could even go further and create helper methods which would expand into corresponding intrinsics like:

```C#

int value = Foo();
Vector<int> vector = value.Shuffle(0);
vector = value.UnpackLow();

// and corresponding vector methods
vector = vector.Shuffle(1);

// what would allow to write
vector = vector.ShiftLeft(4).Shuffle(3).UnpackLow();
```